src/parsers/espresso/7.2/parser.js

```diff
@@ -0,0 +1,292 @@
+    }
+
+    serialize() {
+        const { atomicSpecies } = this.intermediateFormat;
+        const { atomicPositions } = this.intermediateFormat;
+        const { cellParameters } = this.intermediateFormat;
+        if (!cellParameters) {
+            throw new Error("Couldn't read cell parameters");
+            // TODO: add cell generation from ibrav etc...
+        }
+        return { atomicSpecies, atomicPositions, cellParameters };
+    }
+}
+
+/**
+ * @summary checks if the given fileContent is in the format of a Quantum ESPRESSO .in file.
+ * @param {String} fileContent
+ * @return {Boolean}
+ */
+function isEspressoFormat(fileContent) {
+    const espressoRegex = regex.espressoFingerprint;
+    return espressoRegex.test(fileContent);
+}
+
+/**
+ * @summary Return unit cell parameters from CELL_PARAMETERS card
+ * @param {String} text - cards data
+ * @return {{vectors: Number[][], units: String}}
+ */
+function getCellConfig(text) {
+    const match = regex.cellParameters.exec(text);
+    if (match) {
+        const units = match[1];
+        const values = match.slice(2, 11);
+        // creating 3 vectors from 9 values
+        const vectors = Array.from({ length: 3 }, (_, i) =>
+            values.slice(i * 3, i * 3 + 3).map(Number),
+        );
+        return { vectors, units };
+    }
+    throw new Error("Couldn't read cell parameters");
+}
+
+/**
+ * @summary Return units and scaling factor according to Quantum Espresso docs
+ * @param {String} units - units from ATOMIC_POSITIONS card
+ * @returns {{_units: String, scalingFactor: Number}}
+ */
+function convertFromEspressoUnits(units) {
+    let _units, scalingFactor;
+    switch (units) {
+        case "alat":
+            scalingFactor = 1.0;
+            _units = ATOMIC_COORD_UNITS.crystal;
+            break;
+        case "bohr":
+            scalingFactor = coefficients.BOHR_TO_ANGSTROM;
+            _units = ATOMIC_COORD_UNITS.cartesian;
+            break;
+        case "angstrom":
+            scalingFactor = 1.0;
+            _units = ATOMIC_COORD_UNITS.cartesian;
+            break;
+        case "crystal":
+            scalingFactor = 1.0;
+            _units = ATOMIC_COORD_UNITS.crystal;
+            break;
+        case "crystal_sg":
+            throw new Error("crystal_sg not supported yet");
+        default:
+            throw new Error(`Units ${units} not supported`);
+    }
+    return { _units, scalingFactor };
+}
+
+/**
+ * @summary Read atomic positions from ATOMIC_POSITIONS card
+ * @param {String} text - cards data
+ * @returns {{elements: Object[], coordinates: Object[], constraints: Object[], units: String}}
+ */
+function getAtomicPositions(text) {
+    const atomicSpeciesMatches = Array.from(text.matchAll(regex.atomicSpecies));
+    // eslint-disable-next-line no-unused-vars
+    const atomicSpecies = atomicSpeciesMatches.map((match) => ({
+        element: match[1],
+        mass: parseFloat(match[2]),
+        potential: match[3],
+    }));
```

Comment:
Can we remove this? It's unused and the `ATOMIC_POSITIONS` block seems to be enough to define elements, coordinates, and constraints.

Reply:
Ok